pvanalytics/features/daytime.py

```diff
@@ -0,0 +1,247 @@
+        lambda day: any(day)
+    )
+    return _smooth_if_invalid(night, invalid, correction_window)
+
+
+def _filter_and_normalize(series, outliers):
+    # filter a series by removing outliers and clamping the minimum to
+    # 0. Then normalize the series by the maximum deviation.
+    if outliers is not None:
+        series.loc[outliers] = np.nan
+    series.loc[series < 0] = 0
+    return (series - series.min()) / (series.max() - series.min())
+
+
+def _freqstr_to_minutes(freqstr):
+    return pd.to_timedelta(
+        frequencies.to_offset(freqstr)
+    ).seconds / 60
+
+
+def power_or_irradiance(series, outliers=None,
+                        low_value_threshold=0.003,
+                        low_median_threshold=0.0015,
+                        low_diff_threshold=0.0005, median_days=7,
+                        clipping=None, freq=None,
+                        correction_window=31, hours_min=5,
+                        daytime_difference_max=30,
+                        day_length_window=14):
+    """Return True for values that are during the day.
+
+    After removing outliers and normalizing the data, periods of
+    low-slope and low-value are identified as night. A time is
+    classified as night when two of the following three criteria are
+    satisfied:
+
+    - near-zero value
+    - near-zero first-order derivative
+    - near-zero rolling median at the same time over the surrounding
+      week (see `median_days`)
+
+    It is possible that mid-day times where power goes near zero or
+    stops changing can be incorrectly classified as night. To correct
+    these errors times where the total duration of periods marked
+    night or day is too long or too short are identified and values at
+    these times are compared to values at the same time for the
+    preceding two weeks and following two weeks. The day/night value
+    is adjusted to match the median value of these days.
+
+    Finally any values that are True in `clipping` are marked as day.
+
+    Parameters
+    ----------
+    series : Series
+        Time series of power or irradiance.
+    outliers : Series, optional
+        Boolean time series with True for values in `series` that are
+        outliers.
+    low_value_threshold : float, default 0.003
+        Maximum normalized value for a time to be considered night.
+    low_median_threshold : float, default 0.0015
+        Maximum rolling median for a time to be considered night.
+    low_diff_threshold : float, default 0.0005
+        Maximum derivative for a time to be considered night.
+    median_days : int, default 7
+        Number of days to use to calculate the rolling median at each
+        minute.
+    clipping : Series, optional
+        True when clipping indicated. Any values where clipping is
+        indicated are automatically considered 'daytime'.
+    correction_window : int, default 31
+        Number of adjacent days to examine when correcting
+        day/night classification errors.
```

Comment:
May help to note that this should be an odd value

Reply:
I don't think that it has to be odd. If the window is even, the number of values preceding the label will just be one more than the number of following values.